driver/network_driver.go

```diff
@@ -306,3 +312,104 @@ func (d NetworkDriver) ProgramExternalConnectivity(*network.ProgramExternalConne
 func (d NetworkDriver) RevokeExternalConnectivity(*network.RevokeExternalConnectivityRequest) error {
 	return nil
 }
+
+// Try to get the container's labels and update the WorkloadEndpoint with them
+// Since we do not get container info in the libnetwork API methods we need to
+// get then ourselves.
+//
+// This is how:
+// - first we try to get a list of containers attached to the custom network
+// - if there is a container with our endpointID, we try to inspect that container
+// - any labels for that container prefixed by our 'magic' prefix are added to
+//   our WorkloadEndpoint resource
+//
+// Above may take 1 or more retries.
+// internal state and
+func (d NetworkDriver) populateWorkloadEndpointWithLabels(networkID string, endpoint *api.WorkloadEndpoint, dockerCli *dockerClient.Client) {
+	endpointID := endpoint.Metadata.Name
+
+	start := time.Now()
+	deadline := start.Add(5 * time.Second)
```

Comment:
Will add an env var for the deadline. I choose 5s, because I thought it would be enough (yes I remember Bill Gates' famous quote). During my testing I've seen max 1s.